.github/scripts/delete-old-cloudsmith-artifacts.sh

```diff
@@ -1,42 +1,67 @@
 #!/bin/bash
 # Copyright Vespa.ai. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
-dnf install -y 'dnf-command(config-manager)' jq
 
-set -euo pipefail
+if [ "${RUNNER_DEBUG:-}" == "1" ]; then
+    set -o xtrace
+fi
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+dnf install -yq 'dnf-command(config-manager)' jq
+
 MAX_NUMBER_OF_RELEASES=32
+LAST_VESPA_7_RELEASE="7.594.36"
 
 # Cloudsmith repo
 rpm --import 'https://dl.cloudsmith.io/public/vespa/open-source-rpms/gpg.0F3DA3C70D35DA7B.key'
-curl -1sLf 'https://dl.cloudsmith.io/public/vespa/open-source-rpms/config.rpm.txt?distro=el&codename=8' > /tmp/vespa-open-source-rpms.repo
+curl  --silent --show-error --location --fail \
+  --tlsv1 --output /tmp/vespa-open-source-rpms.repo \
+  'https://dl.cloudsmith.io/public/vespa/open-source-rpms/config.rpm.txt?distro=el&codename=8'
 dnf config-manager --add-repo '/tmp/vespa-open-source-rpms.repo'
 rm -f /tmp/vespa-open-source-rpms.repo
 
-# Allow the last Vespa 7 release to remain in the repo
-VERSIONS_TO_DELETE=$(dnf list -y --quiet --showduplicates --disablerepo='*' --enablerepo=vespa-open-source-rpms vespa | awk '/[0-9].*\.[0-9].*\.[0-9].*/{print $2}' | sort -V | grep -v "7.594.36" | head -n -$MAX_NUMBER_OF_RELEASES)
+VESPA_VERSIONS=$(dnf list -y --quiet --showduplicates --disablerepo='*' --enablerepo=vespa-open-source-rpms vespa || true)
+if [[ -z "$VESPA_VERSIONS" ]]; then
+  echo "No Vespa versions found, nothing to do. Exiting."
+  exit 0
+fi
 
+# Note: Allow the last Vespa 7 release to remain in the repo!
+VERSIONS_TO_DELETE=$(echo "${VESPA_VERSIONS}" | awk '/[0-9].*\.[0-9].*\.[0-9].*/{print $2}' | sort -V | grep -v "${LAST_VESPA_7_RELEASE}" | head -n -$MAX_NUMBER_OF_RELEASES)
 if [[ -z "$VERSIONS_TO_DELETE" ]]; then
   echo "No old RPM versions to delete found. Exiting."
   exit 0
 fi
 
 RPMS_TO_DELETE=$(mktemp)
-trap "rm -f $RPMS_TO_DELETE" EXIT
+trap 'rm -f $RPMS_TO_DELETE' EXIT
 
 for VERSION in $VERSIONS_TO_DELETE; do
-  curl -sLf --header 'accept: application/json' \
-    "https://api.cloudsmith.io/v1/packages/vespa/open-source-rpms/?query=version:${VERSION}" | jq -re '.[] | .slug' >> $RPMS_TO_DELETE
+  curl --silent --show-error --location --fail \
+    --header 'accept: application/json' \
+    "https://api.cloudsmith.io/v1/packages/vespa/open-source-rpms/?query=version:${VERSION}" | jq -re '.[] | .slug' >> "${RPMS_TO_DELETE}"
 done
 
-echo "Deleting the following RPMs:"
-cat $RPMS_TO_DELETE
-
+echo
 if [[ "$GITHUB_EVENT_NAME" == "schedule" || "$GITHUB_EVENT_NAME" == "workflow_dispatch" ]]; then
-    for RPMID in $(cat $RPMS_TO_DELETE); do
-      curl -sSLf -X DELETE \
+    echo '\033[1;33mDeleting the following RPMs:\033[0m'
+    echo
+    cat "${RPMS_TO_DELETE}"
```

Comment:
I see you sometimes use curly braces when referencing parameters and sometimes not (when either is semantically equivalent).  If you want my 2 cents I prefer to avoid curly braces as it is easier on the eyes and fingers.

In a somewhat similar vein, I prefer to always reference parameters inside double-quotes (unless that would be semantically wrong), possibly with the exception of integer variables :thinking: .

Reply:
> I see you sometimes use curly braces when referencing parameters and sometimes not (when either is semantically equivalent).

The "sometimes not" is only where I might have forgotten or copied some code from somewhere and not noticed. 
My approach is to use it _always_, for:
- consistency and not need a context switch to quickly identify variables
- using them everywhere possible avoids the need to remember edge cases _(like $9 is ok and $10 isn't)_
- it is consistent with other language templates (python for example)


>  I prefer to always reference parameters inside double-quotes (unless that would be semantically wrong), possibly with the exception of integer variables

That's also my take, if you see around I tend do always quote any usage of variables. Similarly to the `${}`, where they are lacking I probably missed them (unless it was required for the specific usage of the var) or was some code taken from an existing implementation. 😅

Reply:
I think you're in the minority on always wanting to use curly braces.

Reply:
My understanding was that it is best practice. 🙃 

I'll leave this here and tackle this discussion when we look into a linter/code-style as part of [VESPANG-511](https://vespaai.atlassian.net/browse/VESPANG-511).

Reply:
:+1: